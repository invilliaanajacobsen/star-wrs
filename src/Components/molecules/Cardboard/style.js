import styled from 'styled-components';
import ListGroup from 'react-bootstrap/ListGroup'

export const Item = styled(ListGroup.Item)`
	background: #d2a019;
	font-size: 10px;
`
