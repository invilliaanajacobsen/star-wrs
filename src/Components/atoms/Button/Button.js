import React from 'react';
import { ButtonCustom } from './style.js';


const Button = ({ text }) => (
	<ButtonCustom>{text}</ButtonCustom>
);

export default Button;