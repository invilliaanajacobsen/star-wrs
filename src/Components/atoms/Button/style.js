import styled from 'styled-components';
import Button from 'react-bootstrap/Button'


export const ButtonCustom = styled(Button)`
	text-transform: uppercase;
	line-height: 1.3rem;
	border: none;
	border-radius: 6px;
	background-color: #737373;

	:hover {
		background-color: #fff;
		color: #565656;
	  }
`;

