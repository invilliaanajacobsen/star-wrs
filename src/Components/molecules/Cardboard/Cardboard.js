import React, { Fragment, useState } from "react";
import PropTypes from 'prop-types';

import Card from 'react-bootstrap/Card'
import Button from '../../atoms/Button'
import ListGroup from 'react-bootstrap/ListGroup'

import { Item } from "./style.js";


function Cardboard({ title, subtitle, list, button, img }) {

	const [interships, setInterships] = useState(false);

	const showInterships = event => {
		setInterships(true)
	}

	return (
		<Card style={{ width: '48%', height: '100%', margin: 'auto', background: '#d2a019' }}>
			<Card.Img variant="top" src={img} />
			<Card.Body>
				<Card.Title>{title}</Card.Title>
				<Card.Subtitle>{subtitle}</Card.Subtitle>

			</Card.Body>
			{interships && (
				<Fragment>
					<ListGroup variant="flush">
						{list && list.length > 0 ? (
							list.map((data, index) => {

								return (<Item key={`${index}-listGroup`}> {data}</Item>)
							})) : "n√£o possue nave :( "}
					</ListGroup>
				</Fragment>
			)}
			<Card.Body onClick={(e) => showInterships()}>
				<Button text={button} />
			</Card.Body>
		</Card>
	)
}


Cardboard.propTypes = {
	title: PropTypes.string,
	subtitle: PropTypes.any,
};

Cardboard.defaultProps = {
	title: 'Title',
	subtitle: 'Subtitle',
};

export default Cardboard;