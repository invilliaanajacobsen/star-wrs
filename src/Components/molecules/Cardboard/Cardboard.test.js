import React from "react";
import { shallow } from "enzyme";
import Button from '../../atoms/Button'

import Cardboard from "./Cardboard";

var props = {
	title: "title example",
	subtitle: "subtitle example",
	button: "button test",
	list: [1, 2, 3, 4, 5, 6, 7, 8]
}

describe("Card component", () => {

	it("should Card match snapshot with empty content", () => {
		expect(shallow(<Cardboard />)).toMatchSnapshot();
	});

	it("should Card match snapshot with props", () => {
		expect(shallow(<Cardboard {...props} />)).toMatchSnapshot();
	});

	it("should Card match snapshot with props and list", () => {
		expect(shallow(<Cardboard {...props} list={props.list} />)).toMatchSnapshot();
	});

});
