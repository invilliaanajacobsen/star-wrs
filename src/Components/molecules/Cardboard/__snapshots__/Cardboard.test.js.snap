// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should Card match snapshot with empty content 1`] = `
<Card
  body={false}
  style={
    Object {
      "background": "#d2a019",
      "height": "100%",
      "margin": "auto",
      "width": "48%",
    }
  }
>
  <CardImg
    variant="top"
  />
  <CardBody>
    <CardTitle>
      Title
    </CardTitle>
    <CardSubtitle>
      Subtitle
    </CardSubtitle>
  </CardBody>
  <CardBody>
    <Button />
  </CardBody>
</Card>
`;

exports[`Card component should Card match snapshot with props 1`] = `
<Card
  body={false}
  style={
    Object {
      "background": "#d2a019",
      "height": "100%",
      "margin": "auto",
      "width": "48%",
    }
  }
>
  <CardImg
    variant="top"
  />
  <CardBody>
    <CardTitle>
      title example
    </CardTitle>
    <CardSubtitle>
      subtitle example
    </CardSubtitle>
  </CardBody>
  <ListGroup
    variant="flush"
  >
    <ListGroupItem
      active={false}
      disabled={false}
      key="0-listGroup"
      variant={null}
    >
       
      1
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="1-listGroup"
      variant={null}
    >
       
      2
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="2-listGroup"
      variant={null}
    >
       
      3
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="3-listGroup"
      variant={null}
    >
       
      4
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="4-listGroup"
      variant={null}
    >
       
      5
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="5-listGroup"
      variant={null}
    >
       
      6
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="6-listGroup"
      variant={null}
    >
       
      7
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="7-listGroup"
      variant={null}
    >
       
      8
    </ListGroupItem>
  </ListGroup>
  <CardBody>
    <Button
      text="button test"
    />
  </CardBody>
</Card>
`;

exports[`Card component should Card match snapshot with props and list 1`] = `
<Card
  body={false}
  style={
    Object {
      "background": "#d2a019",
      "height": "100%",
      "margin": "auto",
      "width": "48%",
    }
  }
>
  <CardImg
    variant="top"
  />
  <CardBody>
    <CardTitle>
      title example
    </CardTitle>
    <CardSubtitle>
      subtitle example
    </CardSubtitle>
  </CardBody>
  <ListGroup
    variant="flush"
  >
    <ListGroupItem
      active={false}
      disabled={false}
      key="0-listGroup"
      variant={null}
    >
       
      1
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="1-listGroup"
      variant={null}
    >
       
      2
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="2-listGroup"
      variant={null}
    >
       
      3
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="3-listGroup"
      variant={null}
    >
       
      4
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="4-listGroup"
      variant={null}
    >
       
      5
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="5-listGroup"
      variant={null}
    >
       
      6
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="6-listGroup"
      variant={null}
    >
       
      7
    </ListGroupItem>
    <ListGroupItem
      active={false}
      disabled={false}
      key="7-listGroup"
      variant={null}
    >
       
      8
    </ListGroupItem>
  </ListGroup>
  <CardBody>
    <Button
      text="button test"
    />
  </CardBody>
</Card>
`;
